src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -117,12 +118,21 @@ public KillBillHttpClient(final String kbServerUrl, final String username, final
         this.apiKey = apiKey;
         this.apiSecret = apiSecret;
 
-
         final AsyncHttpClientConfig.Builder cfg = new AsyncHttpClientConfig.Builder();
 
+        if(requestTimeout != null)
+        {
+            cfg.setRequestTimeout(requestTimeout);
+            requestTimeoutSec = (int) Math.ceil((double) requestTimeout / 1000);
```

Comment:
If you were to change `requestTimeoutSec` to `requestTimeoutMillis` (and update `executeAndWait` accordingly), you wouldn't need any complex transformation here.